snapcraft/remote/utils.py

```diff
@@ -76,3 +98,57 @@ def _compute_hash(directory: Path) -> str:
 
     all_hashes = "".join(hashes).encode()
     return md5(all_hashes).hexdigest()  # noqa: S324 (insecure-hash-function)
+
+
+def humanize_list(
```

Comment:
This replicates `snapcraft.utils.humanize_list`.

Bigger picture: This is probably a good utility function to put into `craft-application`

Reply:
Based on that, your call.

If remote build is going to become its own library that doesn't depend on `craft-application` (and given how we have this function in so many places), we may even want to make a little `craft-utils` library for all the little miscellany we have. (`humanize_list`, possibly architecture related stuff, maybe even our base error class. But I would want to keep it without non-stdlib dependencies if possible.)

Reply:
craft cli is all about presentation though, utils risks being a dumpster of things we do not know or do not want to think of where to put

Reply:
I decided to leave it as-is.

Reply:
Yeah, that is a concern. `craft-cli` is also a good spot, but I wouldn't want to make `craft-parts` depend on it (and it currently uses its own `humanize_list` in 9 places). Although if we're reducing having it in all the apps + craft-parts to just craft-parts and craft-cli that's already a huge improvement.

I've made this to start out with: https://github.com/canonical/craft-cli/issues/183